### PR TITLE
Uml 1227 fix session redirection

### PR DESCRIPTION
**Pull request by cooperaj**
# Purpose

Attempts to fix our session handling by reworking the logic flow around our various middlewares.

Fixes UML-1227

## Approach

Because we use a redirect to bring the user to a session-expired page we need to ensure that upon re-entry to the site (from the redirect) the users session is still marked as expired even though it wouldn't technically be.

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [ ] ~I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant~
* [x] I have added tests to prove my work
* [ ] ~I have added welsh translation tags and updated translation files~
* [ ] ~I have run an accessibility tool on any pages I have made changes to and fixed any issues found~
* [x] The product team have tested these changes

### CODE REVIEW

File: service-front/app/test/CommonTest/Service/Session/EncryptedCookiePersistenceTest.php [hunk 1]
```diff
@@ -72,18 +71,18 @@ public function setUp()
...
         $this->testKey = new Key('test-id', new EncryptionKey(new HiddenString(random_bytes(32))));
 
-        $this->keyManagerProphecy = $this->prophesize(KeyManagerInterface::class);
+        $this->encrypterProphecy = $this->prophesize(EncryptInterface::class);
 
         // Setup the key manager mock
-        $this->keyManagerProphecy->getEncryptionKey()->willReturn($this->testKey);
-        $this->keyManagerProphecy->getDecryptionKey($this->testKey->getId())->willReturn($this->testKey);
+        //$this->keyManagerProphecy->getEncryptionKey()->willReturn($this->testKey);
+        //$this->keyManagerProphecy->getDecryptionKey($this->testKey->getId())->willReturn($this->testKey);
```
Comment (GemTay) [inline service-front/app/test/CommonTest/Service/Session/EncryptedCookiePersistenceTest.php:78]: i'm assuming this commented code can be removed

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredAttributeAllowlistMiddleware.php [hunk 2]
```diff
@@ -40,11 +42,17 @@ public function __construct(LoggerInterface $logger)
...
+
+        // Ensure that we strip out any session information that shouldn't be in there
+        // if the session has expired.
         /** @var SessionInterface $session */
         $session = $request->getAttribute(SessionMiddleware::SESSION_ATTRIBUTE);
-        if ($session !== null && $session->get(EncryptedCookiePersistence::SESSION_EXPIRED_KEY) !== null) {
+        if ($session !== null && $session->has(EncryptedCookiePersistence::SESSION_EXPIRED_KEY)) {
             $this->stripSession($session);
 
+            // TODO logs incorrect time value as seconds should equal time() - TIME_KEY + Session Length
```
Comment (GemTay) [inline service-front/app/src/Common/src/Middleware/Session/SessionExpiredAttributeAllowlistMiddleware.php:55]: Has this TODO been made into a ticket?

**Review by GemTay — COMMENTED** [review]
All looks great, just some comments

File: service-front/app/test/CommonTest/Service/Session/EncryptedCookiePersistenceTest.php (hunk 1 above)
Comment (cooperaj) [reply service-front/app/test/CommonTest/Service/Session/EncryptedCookiePersistenceTest.php:78]: Ah, needs work that bit. Will likely need removing yeah but I may have rendered the test incomplete.

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredAttributeAllowlistMiddleware.php (hunk 2 above)
Comment (cooperaj) [reply service-front/app/src/Common/src/Middleware/Session/SessionExpiredAttributeAllowlistMiddleware.php:55]: I'll just get this done now.

**Review by SeemaMenon — APPROVED** [review]

**Pull request merged**